src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -970,6 +1068,56 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                     // result = ConditionalSelect(op1, op1Dup, op2Dup)
                     return impSimdAsHWIntrinsicCndSel(clsHnd, retType, baseType, simdSize, op1, op1Dup, op2Dup);
                 }
+
+                case NI_VectorT128_op_Multiply:
+                {
+                    NamedIntrinsic hwIntrinsic = NI_Illegal;
+                    GenTree**      broadcastOp = nullptr;
+
+                    if (varTypeIsArithmetic(op1->TypeGet()))
+                    {
```

Comment:
For Arm64 you don't neet broadcast operation when either `op1` or `op2` are floating-point.
Since the operands will be in SIMD registers already you should use `MultiplyByElement` (where `elementIndex` is `0`) instead

Reply:
I also don't think you need to broadcast an integer value to the whole SIMD register. 
You can use `ins Vd.T[0], Xd` followed by `mul Vd, Vn, Vd.T[0]` instead.

In this case, you can share some of the implementation for floating-point and integer types.

Reply:
We should need a `CreateScalarUnsafe` in this case instead then. I'll update to do the right thing.
* Notably this wasn't possible before the ARM HWIntrinsics were brought online, which is likely why Vector<T> wasn't doing that already.

Reply:
Notably, `MultiplyByScalar` and `MultiplyBySelectedScalar` don't work for `byte`/`sbyte` and so those still need a broadcast.